### PR TITLE
Did Web Validation

### PR DESCRIPTION
**Pull request by andresuribe87**
The method implemented a very naive validation. Now, we return errors when the content-type isn't JSON, or when the body cannot be deserialized to a DID document.

### CODE REVIEW

File: did/web/web.go [hunk 1]
```diff
@@ -32,11 +32,19 @@ const (
...
 // Validate return nil if DID is valid, otherwise the validation error.
 func (d DIDWeb) Validate(ctx context.Context) error {
-	_, err := d.resolveDocBytes(ctx)
+	docBytes, header, err := d.resolveDocBytes(ctx)
 	if err != nil {
 		return errors.Wrap(err, "resolving doc bytes")
 	}
-	return nil
+	contentType := header.Get("Content-Type")
+	if contentType != "application/json" {
```
Comment (decentralgabe) [inline did/web/web.go:40]: extract the content type to a const, consider adding it to a util file
assuming there's no stdlib const

File: did/web/web.go [hunk 2]
```diff
@@ -157,26 +165,28 @@ func (d DIDWeb) Resolve(ctx context.Context) (*did.Document, error) {
...
-		return nil, errors.Wrapf(err, "constructing doc request %+v", docURL)
+		return nil, nil, errors.Wrapf(err, "constructing doc request %+v", docURL)
 	}
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return nil, errors.Wrapf(err, "getting doc %+v", docURL)
+		return nil, nil, errors.Wrapf(err, "getting doc %+v", docURL)
 	}
-	defer resp.Body.Close()
+	defer func(Body io.ReadCloser) {
```
Comment (decentralgabe) [inline did/web/web.go:184]:
```suggestion
	defer func(body io.ReadCloser) {
```

**Review by decentralgabe — APPROVED** [review]
approving given changes

File: did/web/web.go (hunk 1 above)
Comment (andresuribe87) [reply did/web/web.go]: Done.

File: did/web/web.go (hunk 2 above)
Comment (andresuribe87) [reply did/web/web.go]: Done.

**Pull request merged**